src/lib/swapper/swappers/ZrxSwapper/endpoints.ts

```diff
@@ -13,22 +17,61 @@ import type {
   TradeQuote2,
 } from 'lib/swapper/api'
 import { SwapError, SwapErrorType } from 'lib/swapper/api'
-import { isEvmChainAdapter } from 'lib/utils'
+import { assertUnreachable, isEvmChainAdapter } from 'lib/utils'
 
 import { getZrxTradeQuote } from './getZrxTradeQuote/getZrxTradeQuote'
-import type { ZrxSwapStatusResponse } from './types'
 import { fetchZrxQuote } from './utils/fetchZrxQuote'
-import { zrxServiceFactory } from './utils/zrxService'
+
+const createDefaultStatusResponse = (buyTxId: string) => ({
+  status: TxStatus.Unknown,
+  buyTxId,
+  message: undefined,
+})
+
+const tradeQuoteMetadata: Map<string, { chainId: EvmChainId }> = new Map()
+
+const getUnchainedBaseUrl = (chainId: EvmChainId) => {
+  const {
+    REACT_APP_UNCHAINED_AVALANCHE_HTTP_URL,
+    REACT_APP_UNCHAINED_BNBSMARTCHAIN_HTTP_URL,
+    REACT_APP_UNCHAINED_ETHEREUM_HTTP_URL,
+    REACT_APP_UNCHAINED_GNOSIS_HTTP_URL,
+    REACT_APP_UNCHAINED_OPTIMISM_HTTP_URL,
+    REACT_APP_UNCHAINED_POLYGON_HTTP_URL,
+  } = getConfig()
+
+  switch (chainId) {
+    case KnownChainIds.AvalancheMainnet:
+      return REACT_APP_UNCHAINED_AVALANCHE_HTTP_URL
+    case KnownChainIds.BnbSmartChainMainnet:
+      return REACT_APP_UNCHAINED_BNBSMARTCHAIN_HTTP_URL
+    case KnownChainIds.EthereumMainnet:
+      return REACT_APP_UNCHAINED_ETHEREUM_HTTP_URL
+    case KnownChainIds.GnosisMainnet:
+      return REACT_APP_UNCHAINED_GNOSIS_HTTP_URL
+    case KnownChainIds.OptimismMainnet:
+      return REACT_APP_UNCHAINED_OPTIMISM_HTTP_URL
+    case KnownChainIds.PolygonMainnet:
+      return REACT_APP_UNCHAINED_POLYGON_HTTP_URL
+    default:
+      assertUnreachable(chainId)
+  }
+}
```

Comment:
im surprised this doesnt already exists. If your know where i can find it pls link

Reply:
Yeah we don't have it in such form, we have the concepts of adapters, which you can get by ChainId and get the unchained-client instance, but no strict mapping of ChainId to unchained URL